How to create Polymarket clone on Polygon Network

viral-sangani opened the pull request:
Thanks for submitting this tutorial for Figment Learn!

Before you hit "Create Pull Request", make sure you have read the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8) and reviewed the [Sample Tutorial Structure](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d). 

Please confirm the following before submitting your tutorial for review - 

- [x] My tutorial follows the [Tutorial Contribution Guidelines](https://www.notion.so/figmentnetworks/Figment-Learn-Contributor-Guidelines-b08691366e144d33a5636d9276a44ec8).

In particular:

- [x] The title of the Pull Request matches the title of my tutorial.
- [x] I've followed the [tutorial template](https://www.notion.so/figmentnetworks/Sample-Tutorial-Structure-667ac2aad36f4f94a3ebaca053180b2d) structure.
- [x] I've [proofread the tutorial](https://grammarly.com) for spelling and grammar mistakes.
- [x] I've run a linter & prettifier to format my code blocks.
- [x] I've thoroughly explained the code, with inline code comments and text explanations.
- [x] I've run the tutorial to test it and the code works.

**NOTE**: We cannot review your Pull Request if it does not meet the standard outlined above.

pyr0gan reviewed (changes requested):
Hey @viral-sangani 😄 

Thanks for writing this helpful tutorial! There are some fixes to be made, mostly related to code blocks. There are some grammar fixes, and a few cases where breaking the information down into bullet points would be useful.

Please have a look at my comments and make the necessary changes 🙏 Remember to mark the conversations as complete when you've made the changes so that my next review can proceed quickly!

pyr0gan commented on network-documentation/matic/tutorials/how-to-create-polymarket-clone.md:
```diff
@@ -0,0 +1,1676 @@
+
+This is what the dApp we will be creating looks like:
+
+[Demo video](https://vimeo.com/645127735)
```
As I think about it more, it might be good to make a few separate videos to show the functionality being written in the components. That way readers have some visual cues to help them understand what the code is doing. What do you think, @viral-sangani ? 😄

viral-sangani replied:
@pyr0gan I think you are right, few separate videos to show the functionality sounds like a good idea. I will add GIFs after every main component.

pyr0gan commented:
Hi @viral-sangani,

You will need to update this Pull Request to work with the new directory structure of the repository before it can be merged:

- Move images to the `assets` directory, you can then change all image links to the relative path `../assets/image-filename.png`
- Move the markdown file into the `polygon` directory, so the new path would be `polygon/how-to-create-polymarket-clone.md`

viral-sangani commented:
@pyr0gan I have updated the file structure with respect to the new repository structure. 
Also added a few GIFs where needed and shifted the brief dApp demo at the end and added a smaller version at the start.

Thanks

pyr0gan commented:
@viral-sangani my sincere apologies, I was wrong about the image links. It is best to use a direct URL format to avoid any broken images. If you could please change the image links to use `https://github.com/figment-networks/learn-tutorials/raw/master/assets/image-filename.png` instead of the relative path, we can be confident that all images will display properly in the tutorial 😅 .

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+We will start by creating `context.tsx` which will handle the smart contract instances. Create a file `DataContext.tsx` in `context` directory and paste the following code:
+
+```tsx
````
Just for consistency, it would be good to add the comment with filename here as well 😄

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+We will start with `index.tsx`. Open `pages/index.tsx` and paste the following code:
+
+```tsx
````
Comment with filename 😄

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+
+```tsx
+// `components/Navbar.tsx`
````
Great! Just remove the backticks, as they don't render inside of a code block 👍

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+
+```tsx
+// `components/MarketCard.tsx`
````
Just remove the backticks 😄

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+
+```tsx
+// `admin/index.tsx`
````
Remove the backticks 🎉

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+Add the following function in the `Details` component:
+
+```javascript
````
For consistency, add the comment with the path and filename 👍

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+For the graph section, create a new file in the `components/Chart` directory called `ChartContainer.tsx` and paste the following code:
+
+```tsx
````
Comment with path and filename 😄

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+
+```tsx
+// `pages/portfolio.tsx`
````
Remove the backticks 😄

pyr0gan commented on polygon/how-to-create-polymarket-clone.md:
````diff
@@ -0,0 +1,1697 @@
+
+In the Portfolio, we have Navbar, a list of markets, and the total portfolio value. To get the list of markets, paste the following code in the `Portfolio` component before the return statement.
+
+```javascript
````
Comment with path and filename 👍

viral-sangani commented:
@pyr0gan I have updated the links for the images, you can check.
Thanks

pyr0gan reviewed (approved):
Thanks @viral-sangani the changes look good and the added images make it much easier to follow along! 🚀 
Just a couple of quick fixes 👍

viral-sangani commented:
@pyr0gan Made the changes you suggested😀.
